### PR TITLE
Replace hash_map/set with unordered_map/set

### LINKED ISSUE #821: Remove usage of hash_map/hash_set
I changed my buildscripts to generatie VS2015 project files (to fix issue #30) but I found some issues when compiling.
Adding the `_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS` Will probably fix my warming (haven't tried it yet) but I just wanted to let you guys know.

Usage: 
[util.h](https://github.com/dotnet/coreclr/blob/838807429a0828a839958e3b7d392d65886c8f2e/src/ToolBox/SOS/Strike/util.h#L2860)
[decimal.h](https://github.com/dotnet/coreclr/blob/465721555eda899a06bc91fe37bc47c0a1adbdd4/src/debug/daccess/dacimpl.h#L29)

Source: http://blogs.msdn.com/b/vcblog/archive/2014/06/06/c-14-stl-features-fixes-and-breaking-changes-in-visual-studio-14-ctp1.aspx
>In VS14, we're going to deprecate `<hash_map>` and `<hash_set>`.  Including them will emit a static_assert of the form "`<hash_map>` is deprecated and will be REMOVED. Please use `<unordered_map>`. You can define _SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS to acknowledge that you have received this warning."

### PR DESCRIPTION
**Pull request by DickvdBrink**
hash_map/hash_set are deprecated in VS 2015

Haven't found time to fix my issues with building mscorlib on VS2013 so I did not run the tests (jup, i know pretty bad but at least I'm honest about it :-) ).

I did however build on Fedora and VS2015 (but with failures because of the other issues but I could see the hash_map/set issues are now gone).

fixes #821

Let me know if something is wrong and I might be able to fix it :) Thanks!

### CODE REVIEW

File: src/ToolBox/SOS/Strike/gcroot.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@
  *          the GCDesc which is required to walk the object's references.
  *      3.  Use O(1) data structures for anything perf-critical.  Almost all of the data structures we use to
  *          keep track of data have very fast lookups.  For example, to keep track of the objects we've considered
- *          we use a hash_set.  Similarly to keep track of MethodTable data we use a hash_map to track the
+ *          we use a hash_set.  Similarly to keep track of MethodTable data we use a unordered_map to track the
```
Comment (akoeplinger) [inline src/ToolBox/SOS/Strike/gcroot.cpp]: there's another instance of `hash_set` in the comment

Comment (DickvdBrink) [inline src/ToolBox/SOS/Strike/gcroot.cpp]: good catch, did a double check before I created the PR but I think I did something wrong when repushing or something. Sorry and thanks!

**Comment by DickvdBrink** [comment]
There is also some stuff here: https://github.com/dotnet/coreclr/blob/master/src/AUTOEXP.DAT#L901

As I don't know what to do with this.. should I be changed? Or is this oke?

**Comment by jkotas** [comment]
I would leave autoexp.dat alone. It is pretty out of date (most of it is just a copy of the stock file that came with VS2005 - with a couple of CLR specific modification). I do not think anybody is using it these days.

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**